config/confd/templates/haproxy.conf.tmpl

```diff
@@ -1,32 +0,0 @@
-# Copyright (C) 2022 Balena Ltd.
-#
-# This program is free software: you can redistribute it and/or modify
-# it under the terms of the GNU Affero General Public License as published
-# by the Free Software Foundation, either version 3 of the License, or
-# (at your option) any later version.
-#
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU Affero General Public License for more details.
-#
-# You should have received a copy of the GNU Affero General Public License
-# along with this program.  If not, see <https://www.gnu.org/licenses/>.
-
-[Unit]
-Requires=confd.service
-After=confd.service
-
-[Service]
-# Increase the number of sockets haproxy is able to use
-LimitNOFILE=1048576
-# Avoid an issue with failing to mount namespaces
-BindReadOnlyPaths=
-EnvironmentFile=/usr/src/app/config/env
-# wait for open-balena-vpn master to exit
-ExecStop=-/bin/bash -c 'set -x; while curl -sfo/dev/null localhost:8080/ping; do sleep $(( (RANDOM % 3) + 3 ))s; done;'
```

Comment:
Any services previously using this ExecStop should now be a dependency of open-balena-vpn, ensuring that they will not be stopped until open-balena-vpn has fully stopped.

See `config/s6-overlay/s6-rc.d/open-balena-vpn/dependencies.d/`